internal/flink/command_statement_update.go

```diff
@@ -0,0 +1,125 @@
+package flink
+
+import (
+	"fmt"
+
+	"github.com/spf13/cobra"
+
+	pcmd "github.com/confluentinc/cli/v3/pkg/cmd"
+	"github.com/confluentinc/cli/v3/pkg/examples"
+	"github.com/confluentinc/cli/v3/pkg/output"
+	"github.com/confluentinc/cli/v3/pkg/resource"
+)
+
+func (c *command) newStatementUpdateCommand() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:               "update <name>",
+		Short:             "Update a Flink SQL statement.",
+		Args:              cobra.ExactArgs(1),
+		ValidArgsFunction: pcmd.NewValidArgsFunction(c.validStatementArgs),
+		RunE:              c.statementUpdate,
+		Example: examples.BuildExampleString(
+			examples.Example{
+				Text: `Request to update the principal of statement "my-statement" to service account "sa-123456".`,
+				Code: "confluent flink statement update my-statement --principal sa-123456",
+			},
+			examples.Example{
+				Text: `Request to update statement "my-statement" to be run on compute pool "lfcp-123456".`,
```

Comment:
What if the statement is already running? Should we say "Request to move statement..."?